addons/mail/models/mail_activity.py

```diff
@@ -740,7 +740,8 @@ def get_activity_data(self, res_model, domain):
                 'o_closest_deadline': group['date_deadline'],
             }
         activity_type_infos = []
-        activity_type_ids = self.env['mail.activity.type'].search(['|', ('res_model_id.model', '=', res_model), ('res_model_id', '=', False)])
+        activity_type_ids = self.env['mail.activity.type'].sudo().search(
```

Comment:
Note that you break potential ACLs on activity type ... not sure this has a real impact but this sudo changes more than just allowing an access to res_model_id .

Reply:
I am removing (alongside other minor things without much importance) at #74981